File: audace-backend/src/main/java/com/equipe4/audace/model/notification/Notification.java
package com.equipe4.audace.model.notification;

import com.equipe4.audace.dto.notification.NotificationDTO;
import com.equipe4.audace.model.User;
import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
public abstract class Notification {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "SEQUENCE_NOTIFICATION")
    @SequenceGenerator(name = "SEQUENCE_NOTIFICATION", sequenceName = "NOTIFICATION_SEC", allocationSize = 1)
    protected Long id;
    @ManyToOne
    @JoinColumn(name = "user_id")
    protected User user;
    protected NotificationCause cause;

    public abstract NotificationDTO toDTO();

    public enum NotificationCause {
        CREATED,
        UPDATED,
        DELETED,
        EXPIRED
    }
}

File: audace-frontend/src/model/notification.ts
import Application from "./application";
import { CV } from "./cv";
import { Offer } from "./offer";
import { User } from "./user";

export interface Notification {
  id?: number;
  user : User;
  seen : boolean;
  offer? : Offer;
  application? : Application;
  cv? : CV;
  type : String;
  cause : NotificationCause;
}

enum NotificationCause {
  CREATED = "CREATED",
  UPDATED = "UPDATED",
  DELETED = "DELETED",
  EXPIRED = "EXPIRED"
}

export default Notification;
